File: src/node/pickups/create-pickup.js
const Easypost = require('@easypost/api');
const dotenv = require('dotenv');

dotenv.config({ path: '/Users/jhammond/git/easypost/easypost-tools/.env' });
const api = new Easypost(process.env.DEVVM_TEST_API_KEY, {
    baseUrl: "http://oregon1.jhammond.devvm.easypo.net:5000/v2/",
}); // DO NO CREATE IN PROD!

const address = 'adr_a84d92f289e044d9abe91b5b2de42844';
const shipment = 'shp_15ffe591242b4a238f020b3263d9becf';

const pickup = new api.Pickup({
    address,
    shipment,
    reference: 'my-first-pickup',
    min_datetime: '2020-11-20 10:30:00',
    max_datetime: '2020-11-21 10:30:00',
    is_account_address: false,
    instructions: 'Special pickup instructions',
    carrier_accounts: [process.env.DEVVM_DHL_EXPRESS]
});

pickup.save().then(console.log).catch(console.log);

File: src/ruby/pickups/create_pickup.rb
# frozen_string_literal: true

require 'date'
require 'easypost'
require 'dotenv'
require 'amazing_print'

Dotenv.load('/Users/jhammond/git/easypost/easypost-tools/.env')
EasyPost.api_key = ENV['EASYPOST_TEST_API_KEY']

carrier_accounts = [ENV['FEDEX']]

shipment = 'shp_269ac374aa8a4e8893bb1904715577ea'

address = EasyPost::Address.create(
  name: 'EasyPost',
  street1: '417 Montgomery Street',
  street2: '5th Floor',
  city: 'San Francisco',
  state: 'CA',
  zip: '94104',
  country: 'US',
  phone: '4155559999',
  email: 'support@easypost.com'
)

pickup = EasyPost::Pickup.create(
  address: address,
  shipment: shipment,
  reference: 'my-first-pickup',
  min_datetime: DateTime.now,
  max_datetime: DateTime.now + 14_400,
  is_account_address: false,
  instructions: 'Special pickup instructions',
  carrier_accounts: carrier_accounts
)

ap pickup
